src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,172 @@
+    await this.datasource.meetings.syncMeetings();
+  }
+
+  /**
+   * Unregister the sdk meeting plugin from the device.
+   */
+  async disconnect() {
+    await this.datasource.meetings.unregister();
+  }
+
+  /**
+   * Retrieve the local device medias and add them to the meeting
+   * with the given default media settings to the meeting.
+   *
+   * @param {string} meetingID  ID to retrieve the SDK meeting object to add the local media to
+   */
+  addLocalMedia(meetingID) {
+    const meeting = this.fetchMeeting(meetingID);
+
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+
+    meeting.getMediaStreams(mediaSettings).then(([localStream, localShare]) => {
+      meeting.addMedia({
+        localShare,
+        localStream,
+        mediaSettings,
+      });
+    });
+  }
+
+  /**
+   * Update the meeting object with media attached based on a given event type.
+   *
+   * @param {string} ID     ID of the meeting to fetch
+   * @param {Object} media  new media stream to attach to the meeting object based on a given event type
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  attachMedia(ID, {type, stream}) {
+    const meeting = this.meetings[ID];
+
+    switch (type) {
+      case MEDIA_TYPE_LOCAL:
+        this.meetings[ID] = {
+          ...meeting,
+          localAudio: stream.getAudioTracks()[0],
+          localVideo: stream.getVideoTracks()[0],
+        };
+        break;
+      default:
+        this.meetings[ID] = {...meeting};
+        break;
+    }
+  }
+
+  /**
+   * Creates and returns an observable to the new meeting object.
+   *
+   * @param {string} target destination to start the meeting with
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  createMeeting(target) {
+    return from(this.datasource.meetings.create(target)).pipe(
+      map(({id, destination, sipuri}) => {
+        this.meetings[id] = {
+          ID: id,
+          title: sipuri || destination,
+          localVideo: null,
+          localAudio: null,
+          localShare: null,
+          remoteStream: null,
+          remoteShare: null,
+        };
+
+        return this.meetings[id];
+      })
+    );
+  }
+
+  /**
+   * Returns a SDK meeting object retrieved from the collection.
+   *
+   * @param {string} ID ID of the meeting to fetch.
+   * @returns {Meeting}
+   * @memberof MeetingsSDKAdapter
+   * @private
+   */
+  fetchMeeting(ID) {
+    return this.datasource.meetings.getMeetingByType('id', ID);
+  }
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID ID of meeting to get
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsSDKAdapter
+   */
+  getMeeting(ID) {
+    if (!(ID in this.getMeetingObservables)) {
+      const meeting = this.fetchMeeting(ID);
```

Comment:
Since this is an SDK Meeting object and not the observable meeting object, it would be a bit more direct to use this naming:
```suggestion
      const sdkMeeting = this.fetchMeeting(ID);
```